src/Messages.jl

```diff
@@ -103,26 +105,25 @@ Represents a HTTP Response Message.
 You can get each data with [`HTTP.status`](@ref), [`HTTP.headers`](@ref), and [`HTTP.body`](@ref).
 
 """
-mutable struct Response <: Message
+mutable struct Response{T} <: Message
```

Comment:
The `Response` and `Request` objects are now both parameterized on their `body` fields. We previously had some slight awkwardness of having `body` be hard-coded as `Vector{UInt8}`, but then needing to pass a separate `body` argument to all layer functions in case the `body` was an `IO` or some other supported format. We now just parameterize `body` on whatever the user passes and can remove the `body_was_streamed` business and extra argument passing.